osu.Game/Screens/Select/Details.cs

```diff
@@ -0,0 +1,545 @@
+        public BeatmapInfo Beatmap
+        {
+            get
+            {
+                return beatmap;
+            }
+
+            set
+            {
+                if (beatmap == value) return;
+                beatmap = value;
+                description.Text = beatmap.Version;
+                source.Text = beatmap.Metadata.Source;
+                tags.Children = beatmap.Metadata.Tags?.Split(' ').Select(text => new SpriteText
+                {
+                    Text = text,
+                    TextSize = 14,
+                    Font = "Exo2.0-Medium",
+                });
+
+                circleSize.Value = beatmap.Difficulty.CircleSize;
+                drainRate.Value = beatmap.Difficulty.DrainRate;
+                overallDifficulty.Value = beatmap.Difficulty.OverallDifficulty;
+                approachRate.Value = beatmap.Difficulty.ApproachRate;
+                stars.Value = (float) beatmap.StarDifficulty;
+            }
+        }
+
+        private List<int> ratings;
+        public IEnumerable<int> Ratings
+        {
+            get
+            {
+                return ratings;
+            }
+            set
+            {
+                ratings = value.ToList();
+                negativeRatings.Text = ratings.GetRange(0, 5).Sum().ToString();
+                positiveRatings.Text = ratings.GetRange(5, 5).Sum().ToString();
+                ratingsBar.Length = (float)ratings.GetRange(0, 5).Sum() / ratings.Sum();
+
+                List<DetailsBar> ratingsGraphBars = ratingsGraph.Children.ToList();
+                for (int i = 0; i < 10; i++)
+                    if(ratingsGraphBars.Count > i)
+                        ratingsGraphBars[i].Length = (float)ratings[i] / ratings.Max();
+                    else
+                        ratingsGraph.Add(new DetailsBar
+                        {
+                            RelativeSizeAxes = Axes.Both,
+                            Width = 0.1f,
+                            Length = (float)ratings[i] / ratings.Max(),
+                            Direction = BarDirection.BottomToTop,
+                            BackgroundColour = new Color4(0, 0, 0, 0),
+                        });
+            }
+        }
+
+        private List<int> retries = Enumerable.Repeat(0,100).ToList();
+        public IEnumerable<int> Retries
+        {
+            get
+            {
+                return retries;
+            }
+            set
+            {
+                retries = value.ToList();
+                calcRetryAndFailBarLength();
+            }
+        }
+
+        private List<int> fails = Enumerable.Repeat(0,100).ToList();
+        public IEnumerable<int> Fails
+        {
+            get
+            {
+                return fails;
+            }
+            set
+            {
+                fails = value.ToList();
+                calcRetryAndFailBarLength();
+            }
+        }
+
+        private void calcRetryAndFailBarLength()

```

Comment:
This class serves too many purposes for what it is I think. It would be worthwhile to make a new class here that handles everything to do with charts and the calculations happening here internally.
Don't spend too much time making it reusable as graphs are another beast altogether, but it at least needs to be split out from this file or subclassed.